### PR TITLE
Utilise NameOf and String Interpolation

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
All of `\Compilers\VisualBasic\Portable` completed.

### CODE REVIEW

**Comment by AdamSpeight2008** [comment]
You can ignore @bbarry comments as I've kept the original Roslyn implementation

File: src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedMyGroupCollectionPropertyAccessorSymbol.vb [hunk 1]
```diff
@@ -227,15 +230,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
...
-                       "If " & StringConstants.ValueParameterName & " Is " & fieldName & vbCrLf &
-                           "Return" & vbCrLf &
-                       "End If" & vbCrLf &
-                       "If " & StringConstants.ValueParameterName & " IsNot Nothing Then" & vbCrLf &
-                           "Throw New Global.System.ArgumentException(""Property can only be set to Nothing"")" & vbCrLf &
-                       "End If" & vbCrLf &
-                       disposeMethodName & "(Of " & targetTypeName & ")(" & fieldName & ")" & vbCrLf &
-                   "End Set" & vbCrLf
+            Return _
+ $"Set(ByVal {StringConstants.ValueParameterName} As {targetTypeName})
```
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedMyGroupCollectionPropertyAccessorSymbol.vb]: Left aligned as I think the indentation, could cause issues due significant whitespace within the string literal,

File: src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb [hunk 2]
```diff
@@ -640,7 +640,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ''' </summary>
         Friend Shadows Function WithPreviousSubmission(newPreviousSubmission As VisualBasicCompilation) As VisualBasicCompilation
             If Not IsSubmission Then
-                Throw New NotSupportedException("Can't have a previousSubmission when not a submission")
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb]: This is just incorrect.

File: src/Compilers/VisualBasic/Portable/Parser/ParseXml.vb [hunk 3]
```diff
@@ -2192,9 +2191,9 @@ TryResync:
                                                           nameNew,
                                                           node.TextTokens.Node,
                                                           node.QuestionGreaterThanToken)
-            Else
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Parser/ParseXml.vb]: No idea why you changed this.

File: src/Compilers/VisualBasic/Portable/Parser/ParseXml.vb [hunk 4]
```diff
@@ -2320,9 +2311,8 @@ TryResync:
 
             If anyChanges Then
                 Return New XmlNameSyntax(node.Kind, node.GetDiagnostics, node.GetAnnotations, prefix, localName)
-            Else
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Parser/ParseXml.vb]: No idea why you changed this.

File: src/Compilers/VisualBasic/Portable/Parser/ParseXml.vb [hunk 5]
```diff
@@ -2341,9 +2331,8 @@ TryResync:
 
             If anyChanges Then
                 Return New XmlPrefixSyntax(node.Kind, node.GetDiagnostics, node.GetAnnotations, name, colon)
-            Else
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Parser/ParseXml.vb]: No idea why you changed this.

File: src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedMyGroupCollectionPropertyAccessorSymbol.vb [hunk 6]
```diff
@@ -58,12 +58,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
 
             Dim fieldName As String = PropertyOrEvent.AssociatedField.Name
 
-            Dim codeToParse As String =
-                "Partial Class " & containingTypeName & vbCrLf &
-                    "Property " & propertyName & vbCrLf &
-                        GetMethodBlock(fieldName, MakeSafeName(_createOrDisposeMethod), targetTypeName) &
-                    "End Property" & vbCrLf &
-                "End Class" & vbCrLf
+            Dim codeToParse As String = _
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedMyGroupCollectionPropertyAccessorSymbol.vb]: Would prefer some indentation in this string.

File: src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedMyGroupCollectionPropertyAccessorSymbol.vb [hunk 7]
```diff
@@ -169,10 +170,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             '    return <backingField>
             ' End Get
 
-            Return "Get" & vbCrLf &
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedMyGroupCollectionPropertyAccessorSymbol.vb]: Indentation lost.

File: src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedMyGroupCollectionPropertyAccessorSymbol.vb (hunk 1 above)
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedMyGroupCollectionPropertyAccessorSymbol.vb]: Please preserve the indentation.

File: src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedMyGroupCollectionPropertyAccessorSymbol.vb (hunk 7 above)
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedMyGroupCollectionPropertyAccessorSymbol.vb]: @gafter  String Literals have significant whitespace within their double quotes. If the multi-line string literal is indented, indentation is included within the string. Highly likely not what the user intended. See #2231 I think a comment explaining break indentation.

Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedMyGroupCollectionPropertyAccessorSymbol.vb]: I'm referring to the indentation of the body of the Getter being indented relative to the `Get` and `End Get` statements.

**Comment by gafter** [comment]
29 commits is too much for a PR. Also, this PR no longer merges. Please address the current comments, bring it up-to-date, and squash to a single commit. Alternatively if you are no longer interested in offering these changes please close the PR.

**Comment by AdamSpeight2008** [comment]
@gafter I'll resubmit when I get round to it, I'll use this just as a reference to locate the use cases.

**Pull request closed without merging**